### PR TITLE
updates for Isis support

### PR DESCRIPTION
**Pull request by Kelvinrr**

### CODE REVIEW

File: ale/drivers/__init__.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def load(label, props={}, formatter='usgscsm', verbose=False):
     label : str
                String path to the given label file
     """
+    print(props)
+
```
Comment (jessemapel) [inline ale/drivers/__init__.py]: Extra debug print?

File: ale/vis.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    isdjson = json.load(open(isd))
+
+    nlines = isdjson['image_lines']
+    nsamples = isdjson['image_samples']
+
+    # generate meshgrid
+    xs, ys = np.mgrid[0:nsamples:nsamples/nx, 0:nlines:nlines/ny]
+    xs, ys = xs.flatten(), ys.flatten()
+
+    csmcam = knoten.csm.create_csm(isd)
```
Comment (jessemapel) [inline ale/vis.py]: Adds Knoten as dependency of ALE?

File: ale/vis.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    isiscsm_latlondata = pd.DataFrame(isiscsm_latlondata, columns=['isislon', 'isislat', 'csmlon','csmlat', 'difflon', 'difflat', 'magnitude', 'angles'])
+
+    isiscsm_plotlatlon = hv.VectorField((isiscsm_latlondata['isislon'], isiscsm_latlondata['isislat'], isiscsm_latlondata['angles'], isiscsm_latlondata['magnitude']), group='isisvscsmlatlon').opts(opts.VectorField(pivot='tail', colorbar=True, cmap='coolwarm', title='Image2Ground latlon Diff', arrow_heads=True, magnitude='Magnitude', color=dim('Magnitude')))
+    isiscsm_plotlatlon = isiscsm_plotlatlon.redim(x='longitude', y='latitude')
+    isiscsm_plotlatlon = isiscsm_plotlatlon.opts(plot=dict(width=500, height=1000))
+    isiscsm_plotlatlon = isiscsm_plotlatlon*hv.Points(isiscsm_latlondata, ['isislon', 'isislat'], group='isisvscsmlatlon').opts(size=5, tools=['hover'], invert_yaxis=True)
+    ###############################
+
+    return isis2ground2csm_plot, csm2ground2isis_plot, isiscsm_plotlatlon, data, csm2isis_data, isiscsm_latlondata
+
```
Comment (jessemapel) [inline ale/vis.py]: Why is this file in here?

File: notebooks/load.ipynb [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.1"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2
+}
```
Comment (jessemapel) [inline notebooks/load.ipynb]: What's up with this notebook?

File: notebooks/write_MroCtxPds3LabelNaifSpiceDriver.ipynb [hunk 5]
```diff
@@ -160,7 +359,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.3"
+   "version": "3.7.1"
   }
  },
  "nbformat": 4,
```
Comment (jessemapel) [inline notebooks/write_MroCtxPds3LabelNaifSpiceDriver.ipynb]: Why all of the extra output?

File: tests/ctests/CMakeLists.txt [hunk 6]
```diff
@@ -8,12 +8,10 @@ add_executable(runAleTests ${test_source})
...
-                           PRIVATE
-                           ${GSL_INCLUDE_DIRS}
-                           PUBLIC
-                           ${GSL_INCLUDE_DIRS}
-                           ${EIGEN3_INCLUDE_DIR}
-                           ${ALE_INCLUDE_DIRS})
+                            PUBLIC
+                            ${ALE_INCLUDE_DIRS}
+                            ${GSL_INCLUDE_DIRS}
+                            ${EIGEN3_INCLUDE_DIR})
```
Comment (jessemapel) [inline tests/ctests/CMakeLists.txt]: These can be removed and you can just link this as targets the same way they are linked in the top level cmakelist file.

File: ale/vis.py (hunk 3 above)
Comment (Kelvinrr) [reply ale/vis.py]: accidental add

File: notebooks/load.ipynb (hunk 4 above)
Comment (Kelvinrr) [reply notebooks/load.ipynb]: same as vis, was supposed to stay local.

File: src/ale.cpp [hunk 7]
```diff
@@ -367,7 +367,7 @@ namespace ale {
      // Create a Python tuple to hold the arguments to the method.
      PyObject *pArgs = PyTuple_New(3);
      if(!pArgs) {
-       throw runtime_error(getPyTraceback());
+       throw runtime_error("whoops");
```
Comment (acpaquette) [inline src/ale.cpp]: May want to add a better error here. Not sure if the traceback is still viable but that may be something good to replace this with.

File: tests/ctests/CMakeLists.txt [hunk 8]
```diff
@@ -5,15 +5,13 @@ file(GLOB test_source "${CMAKE_SOURCE_DIR}/tests/ctests/*.cpp")
...
-                           ${ALE_INCLUDE_DIRS})
-
+target_link_libraries(runAleTests
+                       ale
+                       GSL::gsl
+                       GSL::gslcblas
+                       Eigen3::Eigen
+                       ${GTEST_LIBRARIES}
+                       ${GTEST_MAIN_LIBRARIES}
+                       pthread)
```
Comment (acpaquette) [inline tests/ctests/CMakeLists.txt]: Any particular reason for making this explicit? Or is this change to fix something else?

File: src/ale.cpp [hunk 9]
```diff
@@ -383,10 +383,10 @@ namespace ale {
...
      // Call the function with the arguments.
      PyObject* pResult = PyObject_CallObject(pFunc, pArgs);
-     Py_DECREF(pArgs);
-     Py_DECREF(pStringFileName);
-     Py_DECREF(pStringProps);
-     Py_DECREF(pStringFormatter);
+     //Py_DECREF(pStringFileName);
+     //Py_DECREF(pStringProps);
+     //Py_DECREF(pStringFormatter);
+     //Py_DECREF(pArgs);
```
Comment (acpaquette) [inline src/ale.cpp]: Are we just not worrying about dereferncing this?

File: include/ale.h [hunk 10]
```diff
@@ -116,7 +119,11 @@ namespace ale {
    *@return
    */
   double interpolate(std::vector<double> points, std::vector<double> times, double time, interpolation interp, int d);
-  std::string load(std::string filename, std::string props="", std::string formatter="usgscsm");
+  std::string loads(std::string filename, std::string props="", std::string formatter="usgscsm", bool verbose=true);
+
+json load(std::string filename, std::string props="", std::string formatter="usgscsm", bool verbose=true);
```
Comment (acpaquette) [inline include/ale.h]: Fix indentation

File: src/ale.cpp [hunk 11]
```diff
@@ -402,8 +397,23 @@ namespace ale {
...
+     Py_DECREF(pResultStr); 
+     Py_DECREF(pStringFileName);
+     Py_DECREF(pStringProps);
+     Py_DECREF(pStringFormatter); 
+     Py_DECREF(pArgs);
 
      return cResult;
  }
+
+ json load(std::string filename, std::string props, std::string formatter, bool verbose) {
```
Comment (acpaquette) [inline src/ale.cpp]: Indentation fix

Comment (Kelvinrr) [reply src/ale.cpp]: That is indented one level as everything else because it exists in a namespace.

File: tests/ctests/CMakeLists.txt (hunk 8 above)
Comment (Kelvinrr) [reply tests/ctests/CMakeLists.txt]: The ale header is simply dropped into the root of the include dirs in the install location, meaning anytime an API change was made, you had to manually delete ale.hpp in the install directory as it was always picked up first as the include resolution order has third-party includes that also exist in the same location, preventing your local ale.hpp from being included. This should always have Ale's local includes first.

File: tests/ctests/CMakeLists.txt [hunk 12]
```diff
@@ -5,15 +5,13 @@ file(GLOB test_source "${CMAKE_SOURCE_DIR}/tests/ctests/*.cpp")
...
-                           ${EIGEN3_INCLUDE_DIR}
-                           ${ALE_INCLUDE_DIRS})
-
+target_link_libraries(runAleTests
+                       ale
+                       GSL::gsl
+                       GSL::gslcblas
+                       Eigen3::Eigen
+                       ${GTEST_LIBRARIES}
+                       ${GTEST_MAIN_LIBRARIES}
```
Comment (jessemapel) [inline tests/ctests/CMakeLists.txt]: This can also just be the gtest target

Comment (jessemapel) [reply tests/ctests/CMakeLists.txt]: nvm, we don't find gtest, we include it. I'm gonna resolve this.

**Pull request merged**